query-projector/src/pull.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use std::collections::{
+    BTreeMap,
+    BTreeSet,
+};
+
+use mentat_core::{
+    Binding,
+    Entid,
+    Schema,
+    StructuredMap,
+    TypedValue,
+    ValueRc,
+};
+
+use mentat_query::{
+    PullAttributeSpec,
+};
+
+use mentat_query_pull::{
+    Puller,
+};
+
+use errors::{
+    Result,
+};
+
+use super::{
+    Index,
+    rusqlite,
+};
+
+#[derive(Clone, Debug)]
+pub(crate) struct PullOperation(pub(crate) Vec<PullAttributeSpec>);
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct PullIndices {
+    pub(crate) sql_index: Index,                   // SQLite row index.
```

Comment:
Are you referring to the data type?  This is all about columns, right, not rows.  I think a comment about what the indices represent here would not go amiss.

Reply:
Sorry, I meant "index into the SQLite row". Changed to say 'column' instead, which is unambiguous.